version: '3.8'

services:
  
  db:
    image: postgres:16-alpine
    restart: always
    environment:
      POSTGRES_DB: monitor_db
      POSTGRES_USER: user
      POSTGRES_PASSWORD: password
    ports:
      - "5432:5432" 
    volumes:
      - db_data:/var/lib/postgresql/data 
    healthcheck: 
      test: ["CMD-SHELL", "pg_isready -U user -d monitor_db"]
      interval: 5s
      timeout: 5s
      retries: 5

  
  backend:
    build:
      context: ./backend
      dockerfile: Dockerfile
    restart: always
    ports:
      - "3000:3000" 
    environment:
      
      DATABASE_URL: postgresql://user:password@db:5432/monitor_db
      JWT_SECRET: your_super_secret_jwt_key_CHANGE_ME # MUDE ISSO!
      PORT: 3000
    depends_on:
      db:
        condition: service_healthy 
    volumes:
      - ./backend:/app 
      - /app/node_modules 
    command: npm run dev 


  frontend:
    build:
      context: ./frontend 
      dockerfile: Dockerfile
    restart: always
    ports:
      - "5173:5173" 
    environment:
      
      VITE_API_URL: http://localhost:3000/api/v1 
    volumes:
      - ./frontend:/app 
      - /app/node_modules 
    command: npm run dev 

volumes:
  db_data: 